src/renderer/components/message/MessageList.tsx

```diff
@@ -409,6 +409,69 @@ export default function MessageList({ accountId, chat, refComposer }: Props) {
     messageListRef.current.scrollTop = messageListRef.current.scrollHeight
   }, [refComposer])
 
+  // This is a fix for
+  // https://github.com/deltachat/deltachat-desktop/issues/3763
+  // That is, when the chat is scrolled to the bottom and you type
+  // a multiline message, the composer would resize and the chat list would
+  // get "scrolled up", i.e. the last message in the chat would get
+  // partically covered. And so if someone sends a message while you're typing,
+  // you'd have scroll down for it manually.
+  //
+  // This also handles other resizes, e.g. quoting a message, attaching a file,
+  // or resizing the Delta Chat window itself.
+  //
+  // The behavior we're implementing here is similar to "scroll anchoring"
+  // that browsers are supposed to perform, but unfortunately simply adding
+  // one scroll anchor at the bottom of the message list doesn't save us
+  // when the scrollable element itself resizes, and not just
+  // the content inside of it.
+  //
+  // A probably better approach would be to use
+  // `flex-direction: column-reverse;`, as in
+  // https://github.com/deltachat/deltachat-desktop/pull/4116,
+  // but it is buggy in Chromium and maybe WebKit (which we'll be using
+  // when we switch to Tauri).
+  //
+  // Should this be `useLayoutEffect`? This appears to be working fine though.
+  useEffect(() => {
+    const el = messageListRef.current
+    if (!el) {
+      return
+    }
+
+    let prevHeight = el.clientHeight
+
+    const observer = new ResizeObserver(entries => {
+      const entry = entries[0]
+      if (!entry) {
+        return
+      }
+      const newHeight = el.clientHeight
+      if (newHeight === prevHeight) {
+        // In case only width changed or something.
+        return
+      }
+
+      const scrollDistanceToBottomBeforeResize =
+        el.scrollHeight - el.scrollTop - prevHeight
+      if (scrollDistanceToBottomBeforeResize < 7) {
```

Comment:
7 should be a named constant, that we can then ensure to be identical to the constant / threshold when the jump down button is shown.